openml/datasets/dataset.py

```diff
@@ -75,7 +76,7 @@ def __init__(self, id=None, name=None, version=None, description=None,
                 logger.debug("Data pickle file already exists.")
             else:
                 try:
-                    data = self._get_arff()
```

Comment:
Format is returned by the dataset xml. In the long run there will be more formats than arff and sparse_arff, so I would prefer to not try to infer this. Plus, the current arff reader can't do this.